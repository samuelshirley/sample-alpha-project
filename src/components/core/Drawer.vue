<template>
  <v-navigation-drawer
    v-model="drawer"
    app
    clipped
    width="200"
  >
    <v-list
      class="pt-0"
      dense
    >
      <v-divider />

      <v-list-tile
        v-for="item in items"
        :key="item.title"
        :to="item.to"
      >
        <v-list-tile-action>
          <v-icon>{{ item.icon }}</v-icon>
        </v-list-tile-action>

        <v-list-tile-content>
          <v-list-tile-title>{{ item.title }}</v-list-tile-title>
        </v-list-tile-content>
      </v-list-tile>
    </v-list>
  </v-navigation-drawer>
</template>

<script>
export default {
  data: () => ({
    drawer: true,
    items: [
      { title: 'Commits', icon: 'mdi-source-commit', to: '/commits' },
    ],
  }),
}
</script>

<style>

</style>
