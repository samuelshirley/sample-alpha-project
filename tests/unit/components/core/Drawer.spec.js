import { mount } from '@vue/test-utils'

import Drawer from '@/components/core/Drawer'

describe('Drawer.vue', () => {
  it('should match snapshot', () => {
    const wrapper = mount(Drawer, {
      stubs: ['router-link'],
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
