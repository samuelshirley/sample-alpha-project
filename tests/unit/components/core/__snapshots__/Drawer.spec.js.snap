// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Drawer.vue should match snapshot 1`] = `
<aside class="v-navigation-drawer v-navigation-drawer--clipped v-navigation-drawer--fixed v-navigation-drawer--is-mobile v-navigation-drawer--open theme--light" style="height: 100%; margin-top: 0px; transform: translateX(0px); width: 200px;">
  <div role="list" class="v-list pt-0 v-list--dense theme--light">
    <hr class="v-divider theme--light">
    <div role="listitem">
      <router-link-stub class="v-list__tile v-list__tile--link theme--light">
        <div class="v-list__tile__action"><i aria-hidden="true" class="v-icon mdi mdi-source-commit theme--light"></i></div>
        <div class="v-list__tile__content">
          <div class="v-list__tile__title">Commits</div>
        </div>
      </router-link-stub>
    </div>
  </div>
  <div class="v-navigation-drawer__border"></div>
</aside>
`;
